File: Algorithmic Toolbox/extras/Passing Cars/passing_cars.py
def passing_cars(array):
    if len(array) <= 1: return 0

    pairs_of_cars = 0
    cars_going_west = 0

    for car in array:
        if car == 0:
            cars_going_west +=1
        else:
            pairs_of_cars +=cars_going_west
            if(pairs_of_cars > 1000000000): return -1
    return pairs_of_cars

if __name__ == "__main__":
    testing_cases = [[0,1,0,1,1],[],[1],[0],[0,0,0,1,1]]

    for i in testing_cases:
        print(passing_cars(i))

File: Algorithmic Toolbox/extras/Passing Cars/passingCars.js


function passingCars(array){
    if(array.length <=1) return 0
    let 
        carsGoingWest = 0,
        pairOfPassingCars = 0
    //tartet is one iteration only ----O(n)
    for( let i in array){
        console.log("i:",i)
        if(arra[i] == 0){//check if its zero
            carsGoingWest +=1
        }else{ //if its not zero, it gotta be one
            pairOfPassingCars += carsGoingWest
            if(pairOfPassingCars > 1000000000) return -1
        }
    }
    return pairOfPassingCars //this is a number, total os cars passing through
}

const testingCases = [[0,1,0,1,1],[],[1],[0],[0,0,0,1,1]]

testingCases.forEach(item => console.log(passingCars(item)))